### PR TITLE
Added support for per-project platform for mx projects.

### PR DESCRIPTION
**Pull request by tzezula**
The mx project are not Ant based, they do not have `PropertyEvaluator`.
Added a new `ProjectPlatform.forProject` method for non Ant based projects.

### CODE REVIEW

File: java/java.project/src/org/netbeans/spi/java/project/support/ProjectPlatform.java [hunk 1]
```diff
@@ -76,28 +75,53 @@ public static JavaPlatform forProject(
...
+        return forProject(owner, jdkHome, platformName, platformType);
+    }
+
+    /**
+     * Creates a transient {@link JavaPlatform} defined in given {@link Project}.
+     * @param owner the project to return the {@link JavaPlatform} for
+     * @param platformHome the {@link JavaPlatform} install folder
+     * @param platformName the {@link JavaPlatform} system name uniquely identifying the platform.
+     * The name has to be valid property name, use the {@link PropertyUtils#getUsablePropertyName(java.lang.String)} to create the name.
+     * For the Ant based project the platform name is a value of {@code platform.active} property.
```
Comment (JaroslavTulach) [inline java/java.project/src/org/netbeans/spi/java/project/support/ProjectPlatform.java]: This sentence is unlikely correct at this place.

Comment (tzezula) [reply java/java.project/src/org/netbeans/spi/java/project/support/ProjectPlatform.java]: I will add apichange.xml.
Regarding the ` For the Ant based project...` I agree, but I have no other place to describe the contract required by Ant project. Ant project's broken project support depends on it. So I prefer to keep it here.

**Comment by neilcsmith-net** [comment]
Thanks for the add.  Fine with me if you want to merge when ready.

**Comment by geertjanw** [comment]
What is an mx project?

**Comment by tzezula** [comment]
The `mx` project is a project type used to develop GraalVM and also used by Ideal Graph Visualiser which is built on top of NetBeans platform.

**Pull request merged**